### PR TITLE
Chore: Update phpunit to 8.5

### PR DESCRIPTION
**Pull request by glensc**

### CODE REVIEW

**Comment by glensc** [comment]
I don't understand how to fix this error:

```

Expectation failed for method name is "insert" when invoked 8 time(s)
Parameter count for invocation MongoCollection::insert(Array (...), Array (...)) is too low.

src/Saver/MongoSaver.php:45
tests/Saver/MongoTest.php:31
```

I've tried as described in docs:
- https://phpunit.readthedocs.io/en/8.5/test-doubles.html#test-doubles-mock-objects-examples-with-consecutive-php

**Comment by luzip665** [comment]
I would like to provide an idea and replace 
`
$collection
            ->expects($this->exactly(count($data)))
            ->method('insert')
            ->withConsecutive(array_map(function ($data) {
                return $this->equalTo($data);
            }, $data));

'
with 
'       $check_data = array();
        foreach ($data as $a) {
            $check_data[] = array($a, ['w' => 0]);
        }
        $collection
            ->expects($this->exactly(count($data)))
            ->method('insert')
            ->withConsecutive($check_data);
`

`array_map` here creates an array with to much entries.
Each entry after the `array_map` contains the value as array, but PHPUnit wants the content of value as array.

**Comment by luzip665** [comment]
I have no glue where is the difference in the phpunit versions, but when I use 

```
        $test_data = array();
        foreach ($data as $a) {
            $test_data[] = array($a, ['w' => 0]);
        }

        $collection
            ->expects($this->exactly(count($data)))
            ->method('insert')
            ->withConsecutive(...$test_data);

```

the error is replaced by 

```
Expectation failed for method name is "insert" when invoked 8 time(s)
Parameter 0 for invocation #0 MongoCollection::insert(Array (...), Array (...)) does not match expected value.
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    '_id' => 'aaaaaaaaaaaaaaaaaaaaaaaa'
+    '_id' => MongoId Object (...)
     'meta' => Array (
         'url' => '/tasks?page=3'
         'simple_url' => '/tasks'
@@ @@
         'get' => Array (...)
         'env' => Array ()
         'SERVER' => Array (...)
-        'request_ts_micro' => Array (...)
+        'request_ts_micro' => MongoDate Object (...)
+        'request_ts' => MongoDate Object (...)
+        'request_date' => '2013-01-21'
     )
     'profile' => Array (...)
 )
```

which seems expectable to me.

The important difference in the call are the `...` in `withConsecutive`.

May be it help.

**Comment by glensc** [comment]
you have omitted `$this->equalTo` calls to items, but I think I've tried too. I'll try again

**Comment by luzip665** [comment]
As far as I see in `vendor/phpunit/phpunit/src/Framework/MockObject/Rule/ConsecutiveParameters.php` the constraint is set to IsEqual, when there is no constraint.

**Comment by glensc** [comment]
Ok, I figured it out: 9eaa7d89!

**Comment by luzip665** [comment]
Just for interest: Would it be possible to check the insert via the MongoSearcher after insertion?
For checking if the data is really inserted and not only the insert-method is called.

**Comment by glensc** [comment]
> Just for interest: Would it be possible to check the insert via the MongoSearcher after insertion? For checking if the data is really inserted and not only the insert-method is called.

https://github.com/perftools/xhgui/pull/358

File: tests/Saver/MongoTest.php [hunk 1]
```diff
@@ -17,9 +17,24 @@ public function testSave(): void
...
             ->getMock();
-        $collection->expects($this->exactly(count($data)))
+
+        $collection
+            ->expects($this->exactly(count($data)))
             ->method('insert')
-            ->withConsecutive($this->equalTo($data));
+            ->withConsecutive(...array_map(function () {
+                    return [
+                        $this->callback(function ($data) {
```
Comment (glensc) [inline tests/Saver/MongoTest.php]: i don't understand how `$this->callback` is able to obtain `$data`, but it works correctly

**Pull request merged**